lib/vega_lite/data.ex

```diff
@@ -327,6 +327,381 @@ defmodule VegaLite.Data do
+  end
+
+  defp polar_plot_data_layers(data, mark, fields, opts) do
+    pi = :math.pi()
+
+    color =
+      case fields[:color] do
+        nil -> []
+        s when is_binary(s) -> [field: s]
+        opts -> opts
+      end
+
+    color_key = color[:field]
+
+    {x_sign, y_sign} = if opts[:direction] == :counter_clockwise, do: {"-", "+"}, else: {"+", "-"}
+
+    rotation = deg_to_rad(opts[:angle_offset])
+
+    x_formula =
+      "datum.x_linear * cos(#{rotation}) #{x_sign}datum.y_linear * sin(#{rotation})"
+
+    y_formula =
+      "#{y_sign}datum.x_linear * sin(#{rotation}) + datum.y_linear * cos(#{rotation})"
+
+    r =
+      case fields[:r] do
+        opts when is_list(opts) -> opts[:field]
+        field when is_binary(field) -> field
+      end
+
+    theta =
+      case fields[:theta] do
+        opts when is_list(opts) -> opts[:field]
+        field when is_binary(field) -> field
+      end
+
+    auto_field_opts =
+      if radius_marks = opts[:radius_marks] do
+        max_radius = Enum.max(radius_marks)
+
+        [
+          scale: [
+            domain: [-max_radius, max_radius]
+          ],
+          axis: [
+            grid: false,
+            ticks: false,
+            domain_opacity: 0,
+            labels: false,
+            title: false,
+            domain: false,
+            offset: 50
+          ]
+        ]
+      else
+        []
+      end
+
+    get_field_opts = fn field ->
+      if opts[:hide_axes] do
+        [type: :quantitative]
+        |> Keyword.merge(fields[field] || [])
+        |> Keyword.merge(auto_field_opts)
+      else
+        Keyword.merge([type: :quantitative], fields[field] || [])
+      end
+    end
+
+    data_fields = if color_key, do: [r, theta, color_key], else: [r, theta]
+
+    auto_tooltip =
+      Enum.map(data, fn
+        {^color_key, _} ->
+          [field: to_string(color_key), type: :nominal]
+
+        {field, _} ->
+          [field: to_string(field), type: :quantitative]
+      end)
+
+    x_opts = get_field_opts.(:x)
+    y_opts = get_field_opts.(:y)
+
+    tooltip =
+      if opts[:radius_marks] do
+        auto_tooltip
+      else
+        auto_tooltip ++ [[field: "x", type: x_opts[:type]], [field: "y", type: y_opts[:type]]]
+      end
+
+    Vl.new()
+    |> Vl.data_from_values(data, only: data_fields)
+    |> Vl.transform(calculate: "datum.#{r} * cos(datum.#{theta} * #{pi / 180})", as: "x_linear")
+    |> Vl.transform(
+      calculate: "datum.#{r} * sin(#{y_sign}datum.#{theta} * #{pi / 180})",
+      as: "y_linear"
+    )
+    |> Vl.transform(calculate: x_formula, as: "x")
+    |> Vl.transform(calculate: y_formula, as: "y")
+    |> encode_mark(mark)
+    |> then(fn vl ->
+      if color_key do
+        Vl.encode_field(vl, :color, color_key,
+          type: :nominal,
+          scale: Keyword.take(color, [:scheme])
+        )
+      else
+        vl
+      end
+    end)
+    |> Vl.encode_field(:x, "x", x_opts)
+    |> Vl.encode_field(:y, "y", y_opts)
```

Comment:
Let's remove x and y sypport for now. The most we can simplify this, the better.